src/ui/include/EditorNS/languagecache.h

```diff
@@ -0,0 +1,102 @@
+#ifndef _LANGUAGECACHE_H_
+#define _LANGUAGECACHE_H_
+
+#include <QList>
+#include <QString>
+#include <QStringList>
+#include <QVector>
+
+/**
+ * @brief Fast and efficient language cache/lookup for Notepadqq
+ */
+namespace EditorNS {
+/**
+ * @brief Language struct containing all the language data
+ */
+struct Language {
+    QString id;
+    QString name;
+    QString mime;
+    QString mode;
+    QStringList fileNames;
+    QStringList fileExtensions;
+    QStringList firstNonBlankLine;
+    Language() {};
+    Language(const Language& o);
+    Language(Language&& o) noexcept;
+    Language& operator=(const Language& o);
```

Comment:
For those assignments you could use `Language& operator=(const Language& o); = default` to save yourself having to define it manually.